src/main/java/org/assertj/core/api/BDDAssumptions.java

```diff
@@ -0,0 +1,2392 @@
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2019 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import java.io.File;
+import java.io.InputStream;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
+import java.util.OptionalLong;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicIntegerArray;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicLongArray;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
+import java.util.concurrent.atomic.AtomicMarkableReference;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.atomic.AtomicReferenceArray;
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+import java.util.concurrent.atomic.AtomicStampedReference;
+import java.util.function.DoublePredicate;
+import java.util.function.IntPredicate;
+import java.util.function.LongPredicate;
+import java.util.function.Predicate;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+import org.assertj.core.api.ThrowableAssert.ThrowingCallable;
+import org.assertj.core.util.CheckReturnValue;
+
+/**
+ * Behavior-driven development style entry point for assumption methods for different types, which allow to skip test execution when assumptions are not met.
+ * <p>
+ * The difference with the {@link Assumptions} class is that entry point methods are named <b><code>given</code></b> instead of
+ * <code>assumeThat</code>.
+ * <p>
+ * {@link BDDAssumptions} and {@link BDDAssertions} complement each other to allow a fluent Behavior-driven development.
+ * <p>
+ * Examples:
+ * <pre>{@code
+ * String frodo = "HOBBIT";
+ * List<String> fellowshipOfTheRing = Arrays.asList("Aragorn", "Gandalf", "Frodo", "Legolas"); // and more
+ * }</pre>
+ * <pre><code class='java'>
+ * {@literal @Test}
+ *  public void given_the_assumption_is_not_met_the_test_is_skipped() {
+ *    given(frodo).isEqualTo("ORC");
+ *
+ *    // ... following code is not executed
+ *    then(fellowshipOfTheRing).contains("Sauron");
+ *  }
+ *
+ * {@literal @Test}
```

Comment:
also `@Test` was not aligned with `public`, see
![image](https://user-images.githubusercontent.com/382613/66199818-0ade5200-e6fc-11e9-8bc6-4126b60c9a6c.png)